src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.cs

```diff
@@ -529,39 +529,32 @@ private void ConvertCurrentMovieToTasproj()
 			MainForm.SetMainformMovieInfo();
 		}
 
-		private bool LoadFile(FileInfo file, bool startsFromSavestate = false, int gotoFrame = 0)
+		private bool LoadFile(FileInfo file, int gotoFrame = 0)
 		{
 			if (!file.Exists)
 			{
 				Settings.RecentTas.HandleLoadError(file.FullName);
 				return false;
 			}
 
+			return LoadMovie((ITasMovie)MovieSession.Get(file.FullName), gotoFrame);
+		}
+
+		private bool LoadMovie(ITasMovie newMovie, int gotoFrame)
+		{
+			if (newMovie == null) return false;
+
 			_engaged = false;
-			var newMovie = (ITasMovie)MovieSession.Get(file.FullName);
 			newMovie.BindMarkersToInput = Settings.BindMarkersToInput;
 			newMovie.TasStateManager.InvalidateCallback = GreenzoneInvalidated;
 
 			if (!HandleMovieLoadStuff(newMovie))
-			{
 				return false;
-			}
 
 			_engaged = true;
 			Settings.RecentTas.Add(newMovie.Filename); // only add if it did load
 
-			if (startsFromSavestate)
-			{
-				GoToFrame(0);
-			}
-			else if (gotoFrame > 0)
-			{
-				GoToFrame(gotoFrame);
-			}
-			else
-			{
-				GoToFrame(CurrentTasMovie.TasSession.CurrentFrame);
-			}

```

Comment:
Would removing this prevent the tasproj from loading and returning the current frame cursor to the old position? (see also: #2181)

(Haven't tested it.)

Reply:
Probably. I couldn't think of any possible use case where that behavior would be desired, and saw no code or comments that appeared to utilize or explain it.

Why would you ever want to re-load the current movie, and want it to automatically seek to the previously loaded frame?

(I certainly can add this back in, it just seemed like a messy bit of code with no explanation whatever so I cleaned it.)

Reply:
I meant when you opened the movie for the first time: the cursor is supposed to remember what frame the movie was on the previous time the movie was saved.

I do think the way it was previously implemented looked a bit messy and could be improved.

Reply:
Oh, that makes sense. I will revert that change.